### PR TITLE
SDI-88: ♻️  More typescript conversion

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: server/routes/login.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+          req.user.username,
+          utilities.createTwoFactorAuthenticationHash(twofactorCode.toString()),
+        )
+
+        if (smsEnabled) {
+          await notify
+            .sendSms(notifySmsTemplate, userAuthentication.Sms || '', {
+              personalisation: { '2fa_code': twofactorCode },
+            })
+            .catch((err: string) => {
```
Comment (petergphillips) [inline server/routes/login.ts:104]: dubious about this, think `err` should be of type `Error` but that's what the existing code is doing

File: server/@types/notifyClient/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+declare module 'notifications-node-client' {
+  export default class NotifyClient {
+    constructor(url: string, clientKey: string): NotifyClient
```
Comment (petergphillips) [inline server/@types/notifyClient/index.d.ts:3]: this isn't quite right.  I get the error:
```
[Node] TypeError: notifications_node_client_1.default is not a constructor
[Node]     at loginRouter (/Users/peter.phillips/work/check-my-diary/dist/server/routes/login.js:32:26)
```
when the application now runs.  Not sure if I should have declared an interface instead?

Comment (psoleckimoj) [reply server/@types/notifyClient/index.d.ts:3]: @petergphillips I've replied, did a PR and pulled the type file someone had done in a PR on the project itself, seems to work 👍

**Review by mikehalmamoj — APPROVED** [review]

File: server/@types/notifyClient/index.d.ts (hunk 2 above)
Comment (steverendell) [reply server/@types/notifyClient/index.d.ts:17]: this will all get removed soon anyway ?

**Pull request merged**